app/src/main/java/com/itssuryansh/taaveez/Add_New_Content.kt

```diff
@@ -81,12 +101,57 @@ class Add_New_Content : AppCompatActivity() {
         binding?.btnUndo?.setOnClickListener {
             PoemDes?.undo()
         }
+        binding?.btnAddLabel?.setOnClickListener {
+            val dialogView = LayoutInflater.from(this).inflate(R.layout.diaglog_insert_label, null)
```

Comment:
rather than instantiating the dialog, setting up the listeners, and else each and every time `btnAddLabel` is clicked, why don't you do the instantiation, setup and all only once, and on button press just display it and perform only the actions required specifically on button press.